### PR TITLE
New Resource: aws_medialive_channel

### LINKED ISSUE #4936: feature request: medialive resources
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Currently there is not a resource available for Elemental MediaLive. Looking to have the ability to specify input and output groups, input specifications (maximum input bitrate, etc.), destinations and security groups.

### New or Affected Resource(s)

* aws_elemental_medialive

### PR DESCRIPTION
**Pull request by johnsonaj**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #4936

Output from acceptance testing:

```console
$ make testacc TESTARGS='-run=TestAccMediaLiveChannel_' PKG=medialive

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/medialive/... -v -count 1 -parallel 20  -run=TestAccMediaLiveChannel_ -timeout 180m
=== RUN   TestAccMediaLiveChannel_basic
=== PAUSE TestAccMediaLiveChannel_basic
=== RUN   TestAccMediaLiveChannel_update
=== PAUSE TestAccMediaLiveChannel_update
=== RUN   TestAccMediaLiveChannel_updateTags
=== PAUSE TestAccMediaLiveChannel_updateTags
=== RUN   TestAccMediaLiveChannel_disappears
=== PAUSE TestAccMediaLiveChannel_disappears
=== CONT  TestAccMediaLiveChannel_basic
=== CONT  TestAccMediaLiveChannel_updateTags
=== CONT  TestAccMediaLiveChannel_disappears
=== CONT  TestAccMediaLiveChannel_update
--- PASS: TestAccMediaLiveChannel_disappears (70.62s)
--- PASS: TestAccMediaLiveChannel_basic (72.52s)
--- PASS: TestAccMediaLiveChannel_updateTags (100.18s)
--- PASS: TestAccMediaLiveChannel_update (120.43s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/medialive	123.006s

...
```

### CODE REVIEW

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
make testacc TESTARGS='-run=TestAccMediaLiveChannel_' PKG=medialive
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/medialive/... -v -count 1 -parallel 20  -run=TestAccMediaLiveChannel_ -timeout 180m
=== RUN   TestAccMediaLiveChannel_basic
=== PAUSE TestAccMediaLiveChannel_basic
=== RUN   TestAccMediaLiveChannel_update
=== PAUSE TestAccMediaLiveChannel_update
=== RUN   TestAccMediaLiveChannel_updateTags
=== PAUSE TestAccMediaLiveChannel_updateTags
=== RUN   TestAccMediaLiveChannel_disappears
=== PAUSE TestAccMediaLiveChannel_disappears
=== CONT  TestAccMediaLiveChannel_basic
=== CONT  TestAccMediaLiveChannel_updateTags
=== CONT  TestAccMediaLiveChannel_disappears
=== CONT  TestAccMediaLiveChannel_update
--- PASS: TestAccMediaLiveChannel_basic (86.89s)
--- PASS: TestAccMediaLiveChannel_disappears (99.91s)
--- PASS: TestAccMediaLiveChannel_updateTags (118.52s)
--- PASS: TestAccMediaLiveChannel_update (175.77s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/medialive  178.642s
```

File: internal/service/medialive/channel.go [hunk 1]
```diff
@@ -0,0 +1,1435 @@
...
+		return create.DiagError(names.MediaLive, create.ErrActionWaitingForDeletion, ResNameChannel, d.Id(), err)
+	}
+
+	return nil
+}
+
+func waitChannelCreated(ctx context.Context, conn *medialive.Client, id string, timeout time.Duration) (*medialive.DescribeChannelOutput, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending:                   enum.Slice(types.ChannelStateCreating),
+		Target:                    enum.Slice(types.ChannelStateIdle),
```
[the remark this answers is not in the document]
Comment (johnsonaj) [reply internal/service/medialive/channel.go:912]: @jar-b not in state for this waiter. Those states would work when this resource has the ability to start a channel. We can then create  waiters for `waitChannelStarted` and `waitChannelStopped`

File: internal/service/medialive/channel.go [hunk 2]
```diff
@@ -0,0 +1,1435 @@
...
+		return out, err
+	}
+
+	return nil, err
+}
+
+func waitChannelUpdated(ctx context.Context, conn *medialive.Client, id string, timeout time.Duration) (*medialive.DescribeChannelOutput, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending:                   enum.Slice(types.ChannelStateUpdating),
+		Target:                    enum.Slice(types.ChannelStateIdle),
```
[the remark this answers is not in the document]
Comment (johnsonaj) [reply internal/service/medialive/channel.go:930]: @jar-b similar to the the other waiter. To updated we would first need to stop the channel

File: internal/service/medialive/channel.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jar-b) [reply internal/service/medialive/channel.go:912]: got it - thanks for the context! 👍

**Pull request merged**